src/components/SignIn.tsx

```diff
@@ -0,0 +1,77 @@
+import { Grid,Paper,Avatar, TextField,Button} from "@mui/material";
+import  React, { useState } from "react";
+import {useNavigate} from 'react-router-dom';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+
+
+const SignIn = () => {
+
+    //useNavigate is equle to navigate variable
+    const navigate = useNavigate();
+
+    //navigate to home using a function
+    const navigateToHome = () =>{
+        navigate('/home');
+    }
+    
+    //add the data using array for useSate 
+    const [names,setNames] = useState(['sip','usra','udy','baji']);
+    // for select a name in array
+    const [selectedName,setSelectedName] = useState('');
+
+    // this event handler use for random btn
+    const handleClick = () => {
+        //get the indexnum in the 'names' array
+        const randomIndex = Math.floor(Math.random() * names.length);
```

Comment:
move this random name-generating function to a util file inside a folder named utility